os/fs/smartfs/smartfs_utils.c

```diff
@@ -2922,7 +2975,7 @@ static int smartfs_set_transaction(struct smartfs_mountpt_s *fs, uint16_t sector
 		/* Read current current transaction info */
 		ret = FS_IOCTL(fs, BIOC_READSECT, (unsigned long)&req);
 		if (ret < 0) {
-			return ERROR;
+			return ret;
```

Comment:
Is there a reason to return ret instead of ERROR?

Reply:
Yes, by returning "ret", we can know the error type of "FS_IOCTL" in detail.